core/src/test/java/org/apache/brooklyn/util/core/task/ValueResolverTest.java

```diff
@@ -124,4 +133,141 @@ public void testDefaultBeforeDelayAndError() {
         Assert.assertEquals(result.get(), "foo");
     }
 
+    public void testGetImmediately() {
+        MyImmediateAndDeferredSupplier supplier = new MyImmediateAndDeferredSupplier();
+        CallInfo callInfo = Tasks.resolving(supplier).as(CallInfo.class).context(app).immediately(true).get();
+        assertNull(callInfo.task);
+        assertContainsCallingMethod(callInfo.stackTrace, "testGetImmediately");
+    }
+    
+    public void testGetImmediateSupplierWithTimeoutUsesBlocking() {
+        MyImmediateAndDeferredSupplier supplier = new MyImmediateAndDeferredSupplier();
+        CallInfo callInfo = Tasks.resolving(supplier).as(CallInfo.class).context(app).timeout(Asserts.DEFAULT_LONG_TIMEOUT).get();
+        assertNotNull(callInfo.task);
+        assertNotContainsCallingMethod(callInfo.stackTrace, "testGetImmediately");
```

Comment:
The test is called `testGetImmediateXXXX` not `testGetImmediatelyXXXX`.  In any case it might be worth supplying the full method name here to avoid possible confusion with line 140. Same with line 170 below.